### ISSUE TITLE
Support Cancellation

### ISSUE THREAD

**Issue by jdneo** [microsoft/build-server-for-gradle#58]
The Cancellation in BSP is inherited from LSP. See:

https://github.com/build-server-protocol/build-server-protocol/issues/524

#### Cancellation in LSP
https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#cancelRequest
https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#errorCodes

#### How LSP node client sending the cancel request
https://github.com/microsoft/vscode-languageserver-node/blob/f2ff7d55464a1f58f978cb6635bd8865f050553c/jsonrpc/src/common/connection.ts#L1398-L1412

**Comment by jdneo** [comment]
@Sidhaarthsr BTW, just an ask. Are you interested in this issue?

**Comment by Sidhaarthsr** [comment]
Yes @jdneo I'm interested and it's part of this [initiative](https://gitlab.eclipse.org/eclipsefdn/emo-team/gsoc-at-the-ef/-/issues/5). I can pick this up

**Comment by jdneo** [comment]
Thank you :)
